service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -292,6 +296,108 @@ private Span createSpan(final String name, final SpanContext parent, final Tenan
+        final long periodInDays = Optional.ofNullable(connectionDurationConfig.getPeriod())
+                .map(ResourceLimitsPeriod::getNoOfDays)
+                .orElse(0);
+
+        log.trace("connection duration config for the tenant [{}] is [{}:{}, {}:{}, {}:{}, {}:{}]",
+                tenant.getTenantId(),
+                TenantConstants.FIELD_MAX_MINUTES, maxConnectionDurationInMinutes,
+                TenantConstants.FIELD_EFFECTIVE_SINCE, effectiveSince,
+                TenantConstants.FIELD_PERIOD_MODE, periodMode,
+                TenantConstants.FIELD_PERIOD_NO_OF_DAYS, periodInDays);
+
+        if (maxConnectionDurationInMinutes == TenantConstants.UNLIMITED_MINUTES || effectiveSince == null || PeriodMode.UNKNOWN.equals(periodMode)) {
+            result.complete(Boolean.FALSE);
+        } else {
+            final long allowedMaxMinutes = getOrAddToCache(limitsCache,
+                    String.format("%s_allowed_max_minutes", tenant.getTenantId()),
+                    () -> calculateEffectiveLimit(
+                            OffsetDateTime.ofInstant(effectiveSince, ZoneOffset.UTC),
+                            OffsetDateTime.now(ZoneOffset.UTC),
+                            periodMode,
+                            maxConnectionDurationInMinutes));
+            final long connectionDurationUsagePeriod = getOrAddToCache(limitsCache,
+                    String.format("%s_conn_duration_usage_period", tenant.getTenantId()),
+                    () -> calculateResourceUsagePeriod(
+                            OffsetDateTime.ofInstant(effectiveSince, ZoneOffset.UTC),
+                            OffsetDateTime.now(ZoneOffset.UTC),
+                            periodMode,
+                            periodInDays));
+
+            items.put("current period connection duration limit in minutes", allowedMaxMinutes);
+
+            if (connectionDurationUsagePeriod <= 0) {
+                result.complete(Boolean.FALSE);
+            } else {
+                final String queryParams = String.format("minute( sum (increase (%s {tenant=\"%s\"} [%sd])))",
+                        CONNECTIONS_DURATION_METRIC_NAME,
+                        tenant.getTenantId(),
+                        connectionDurationUsagePeriod);
+                final String key = String.format("%s_minutes_consumed", tenant.getTenantId());
+
+                Optional.ofNullable(limitsCache)
+                        .map(ok -> limitsCache.get(key))
+                        .map(cachedValue -> Future.succeededFuture((long) cachedValue))
+                        .orElseGet(() -> executeQuery(queryParams, span)
+                                .map(minutesConnected -> addToCache(limitsCache, key, minutesConnected)))
+                        .map(minutesConnected -> {
+                            items.put("current period connection duration limit in minutes", minutesConnected);
```

Comment:
this is not a *limit* but simply the current period's connection duration, isn't it?

Reply:
yes, I have updated it and pushed the change.